test/metabase/actions/test_util.clj

```diff
@@ -103,11 +103,18 @@
   `(do-with-dataset-definition actions-test-data (fn [] ~@body)))
 
 (defmacro with-temp-test-data
-  "Sets the current dataset to a freshly created dataset-definition that gets destroyed at the conclusion of `body`.
-   Use this to test destructive actions that may modify the data."
+  "Sets the current dataset to a freshly created table-definitions that gets destroyed at the conclusion of `body`.
+   Use this to test destructive actions that may modify the data.
+    (with-temp-test-data [[\"product\"
+                           [{:field-name \"name\" :base-type :type/Text}]
+                           [[\"Tesla Model S\"]]]
+                          [\"rating\"
+                           [{:field-name \"score\" :base-type :type/Integer}]
+                           [[5]]]]
+      ...)"
   {:style/indent :defn}
-  [dataset-definition & body]
-  `(do-with-dataset-definition (tx/dataset-definition ~(str (gensym)) ~dataset-definition) (fn [] ~@body)))
+  [table-definitions & body]
+  `(do-with-dataset-definition (tx/dataset-definition ~(str (gensym)) ~@table-definitions) (fn [] ~@body)))
```

Comment:
fyi my version uses `apply` so the `table-definitions` can be calculated without needing to be a literal list